modules/opensimplex/noise_texture.cpp

```diff
@@ -73,7 +72,8 @@ void NoiseTexture::_bind_methods() {
 	ClassDB::bind_method(D_METHOD("_generate_texture"), &NoiseTexture::_generate_texture);
 	ClassDB::bind_method(D_METHOD("_thread_done", "image"), &NoiseTexture::_thread_done);
 
-	ADD_PROPERTY(PropertyInfo(Variant::VECTOR2, "size"), "set_size", "get_size");
+	ADD_PROPERTY(PropertyInfo(Variant::INT, "width", PROPERTY_HINT_RANGE, "1,2048,1,or_greater"), "set_width", "get_width");
```

Comment:
Aren't `width` and `height` redundant too? `Texture` has them already.

Reply:
Well it only has getters though, not setters.

Reply:
Yes, I need the setters to update the texture when either width or height changes.